### ISSUE TITLE
Ability to get full path of File object

### ISSUE THREAD

**Issue by alex-tee** [mesonbuild/meson#8433]
when you have this
```meson
my_files = files(['myfile1.scm', 'myfile2.scm'])
```

you should be able to do this
```meson
foreach f : my_files
  message (f.full_path())
endforeach
```

so you can reuse those paths elsewhere

**Comment by viric** [comment]
full path or even the path relative to the src root. The object could have multiple methods.

**Comment by tristan957** [comment]
I started work on a PR to implement this fyi :)

**Comment by randy408** [comment]
I have a project where files can be reloaded at runtime when debugging, the files are also part of the build so it just makes sense to generate the source<->fullpath associations from meson for change monitoring and reloading.

**Comment by bruchar1** [comment]
Fixed in #12408.
